### PR TITLE
TT-9957 Fix timeouts when getting keys when mdcb is down

### PR DESCRIPTION
**Pull request by buger**
- Have a JWT token with client_id claim
- Start edge gateway, MDCB disconnected, try to use JWT token with client_id claim, it can be a bit slow first time, but overall works well
- Start edge gateway with MDCb connected, and turn off MDCB in the middle, same token hangs

It happens because `IsClientConnected` function do not respect emergency mode. Now it does. 

Additionally I added a 1 second cache for failed GetKey attempts (or when key not found), to the RPC GetRawKey function, which should signiciantly improve latency for cases like when JWT is used with client_id scope, but not actually use DCR, and such client_id in Tyk does not exits.

### CODE REVIEW

**Comment by buger** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5574/merge
Commit:  
Triggered by: pull_request (@buger)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/6356808147)

**Comment by buger** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5574/merge
Commit: 611520cea6e3579c55d68b4d40850abbe3e14ddd 
Triggered by: pull_request (@buger)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/6356866770)

File: gateway/rpc_storage_handler.go [hunk 1]
```diff
@@ -259,27 +261,22 @@ func (r *RPCStorageHandler) GetRawKey(keyName string) (string, error) {
...
+			// Errors, and key not found, should be cached for a small amount of time
+			cacheStore.Set(keyName, err, 1)
 		}
-		log.Debug("Error trying to get value:", err)
 		return "", storage.ErrKeyNotFound
 	}
-	if r.Gw.GetConfig().SlaveOptions.EnableRPCCache {
-		// Cache key
-		cacheStore := r.Gw.RPCGlobalCache
-		if strings.Contains(keyName, "cert-") {
```
Comment (tbuchaillot) [inline gateway/rpc_storage_handler.go:277]: @buger why this was removed?

Comment (tbuchaillot) [reply gateway/rpc_storage_handler.go:277]: nevermind - it was duplicated code

**Comment by buger** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5574/merge
Commit: fc2c42c50e3966c4ce137a5f054f3f40be5cf6f6 
Triggered by: pull_request (@tbuchaillot)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/6380869151)

**Review by tbuchaillot — APPROVED** [review]
LGTM

**Comment by buger** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5574/merge
Commit: b224efddc373e76f6345bd94c363aa1b100b1e51 
Triggered by: pull_request (@tbuchaillot)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/6391264640)

**Comment by buger** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5574/merge
Commit: 117676e62252de99ba6355798352209b87cfbe58 
Triggered by: pull_request (@tbuchaillot)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/6391388935)

**Comment by buger** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5574/merge
Commit: 751b44e2f94abef8aac4e7e49cf138ab4e8b67cc 
Triggered by: pull_request (@tbuchaillot)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/6391503814)

**Pull request merged**